src/sciline/pipeline.py

```diff
@@ -194,3 +199,93 @@ def bind_and_call(
     def _repr_html_(self) -> str:
         nodes = ((key, data) for key, data in self._graph.nodes.items())
         return pipeline_html_repr(nodes)
+
+
+def get_mapped_node_names(
+    graph: Pipeline, key: type, index_names: Sequence[Hashable] | None = None
+) -> pandas.Series:
+    """
+    Given a graph with key depending on mapped nodes, return a series corresponding
+    mapped keys.
+
+    This is meant to be used in combination with :py:func:`Pipeline.map`.
+    If the key depends on multiple indices, the series will be a multi-index series.
```

Comment:
This took me a while to understand. How about this?
```suggestion
    Return all mapped keys for a base key.

    Given a graph with mapped nodes and a key to a node, this function returns
    the keys of mapped nodes that correspond to the key.
    This is meant to be used in combination with :py:func:`Pipeline.map`.
    If the key depends on multiple indices, the series will be a multi-index series.
```
And an example would go a long way to explaining the function.